src/Chatkit.php

```diff
@@ -393,17 +397,21 @@ protected function getServerToken($user_id = null)
      */
     protected function execCurl($ch)
     {
-        $response = array();
+        $response = json_decode(curl_exec($ch), true);
+        $status = curl_getinfo($ch, CURLINFO_HTTP_CODE);
 
-        $response['body'] = curl_exec($ch);
-        $response['status'] = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+        // inform the user of a connection failure
+        if ($status == 0 || $response === false) {
+            throw new ConnectionException(curl_error($ch));
+        }
 
-        if ($response['body'] === false || $response['status'] < 200 || 400 <= $response['status']) {
-            $this->log('ERROR: execCurl error: '.curl_error($ch));
+        // or an error response from Chatkit
+        if ($status >= 400) {
+            $this->log('ERROR: execCurl error: '.print_r($response, true));
```

Comment:
This is looks good for now, however we return a hash that is something like:
```
{
  "error_type": <Type of error>,
  "error_description": <Error desc>,
  "error_uri": <Documentation URI>,
}
```

Might be better to return all the parts?

Reply:
@vivangkumar The `ChatkitException` which is thrown *does* contain all the parts.

Reply:
Oh, it does. Ignore me.